### PR TITLE
Test against kubernetes 1.18 release

### PR DESCRIPTION
**Pull request by lilic**
Only test against 1.18 release of kubernetes in master.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,15 +89,16 @@ $ minikube addons disable metrics-server
...
-|-----------------------|-----------------|-----------------|-----------------|-----------------|
-| `release-0.3`         | ✔              | ✔              | ✔              | ✔              |
-| `release-0.4`         | ✗              | ✗              | ✔              | ✔              |
-| `HEAD`                | ✗              | ✗              | ✗              | ✔              |
+| kube-prometheus stack | Kubernetes 1.14 | Kubernetes 1.15 | Kubernetes 1.16 | Kubernetes 1.17 | Kubernetes 1.18 |
+|-----------------------|-----------------|-----------------|-----------------|-----------------|-----------------|
+| `release-0.3`         | ✔              | ✔              | ✔              | ✔              | ✗
+| `release-0.4`         | ✗              | ✗              | ✔              | ✔              | ✗
+| `release-0.5`         | ✗              | ✗              | ✗              | ✔              | ✔
+| `HEAD`                | ✗              | ✗              | ✗              | ✗              | ✔
```
Comment (dgrisonnet) [inline README.md:97]: Do we really want to drop support for `1.17` in master?

Comment (lilic) [reply README.md:97]: I think its not dropping support just more dropping testing this version. I can rephrase the compatibility matrix to include a maybe it works 🤷‍♀️

Comment (lilic) [reply README.md:97]: "We know these are supported and work as we test against these versions, but other versions might work." WDYT?

File: tests/e2e/travis-e2e.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ chmod +x kubectl
 curl -Lo kind https://github.com/kubernetes-sigs/kind/releases/download/v0.6.1/kind-linux-amd64
 chmod +x kind
 
-./kind create cluster --image=kindest/node:v1.17.0
+./kind create cluster --image=kindest/node:v1.18.1
```
Comment (dgrisonnet) [inline tests/e2e/travis-e2e.sh:16]: You should use `v1.18.0` instead. The 1.18.1 image as not yet been pushed to [dockerhub](https://hub.docker.com/r/kindest/node/tags?page=1&name=1.18).

Comment (lilic) [reply tests/e2e/travis-e2e.sh:16]: This is why I like minikube more :D sorry kind folks

File: README.md (hunk 1 above)
Comment (dgrisonnet) [reply README.md:97]: Sounds good to me

Comment (brancz) [reply README.md:97]: sgtm

**Comment by brancz** [comment]
lgtm

**Comment by lilic** [comment]
PR is failing due to new kind version, having a look 👀

**Comment by lilic** [comment]
@brancz @dgrisonnet PTAL, had to fix some things as kind broke their compatibility and older version of kind did not work with 1.18 k8s version.

**Comment by brancz** [comment]
lgtm :tada:

**Comment by paulfantom** [comment]
IDEA: can we test always against stable version just by getting version from `https://storage.googleapis.com/kubernetes-release/release/stable.txt`?

**Comment by dgrisonnet** [comment]
That would be great; however, since `kind` does not follow Kubernetes release process, I would say we can't. For example, the latest kind image release is `v1.18.0`.
Do you see any alternatives?

**Comment by lilic** [comment]
*cough* minikube *cough* but @brancz doesn't like it :p

**Comment by lilic** [comment]
But yes, agreed, latest would be ideal.

**Pull request merged**